src/weather.c

```diff
@@ -1,15 +1,196 @@
 #include "global.h"
 #include "weather.h"
 
+#include "constants/ability.h"
 #include "constants/item.h"
+#include "constants/move_id.h"
+#include "constants/type.h"
 #include "constants/weather.h"
 #include "dungeon_global_data.h"
 #include "dungeon_items.h"
 #include "dungeon_random.h"
+#include "dungeon_leader.h"
+#include "dungeon_util.h"
+#include "code_80521D0.h"
+#include "dungeon_ai.h"
+#include "dungeon_pokemon_attributes.h"
+#include "moves.h"
+#include "move_util.h"
 
 extern void TriggerWeatherAbilities(void);
 extern void sub_807EAA0(u32, u32);
 void sub_807E5E4(u8);
+extern u8 UseAttack(struct Entity *);
+extern void sub_805E804(void);
+void sub_806A2BC(struct Entity *pokemon, u8 param_2);
+extern u8 sub_8044B28(void);
+extern void sub_80426C8(u32, u32);
+extern void sub_8068FE0(struct Entity *, u32, struct Entity *r2);
+extern void nullsub_93(struct Position *);
+extern void sub_80522F4(struct Entity *, struct Entity *, u8 *);
+extern void SetMessageArgument(char[], struct Entity*, u32);
+extern void sub_806F370(struct Entity *r0, struct Entity *r1, u32, u32, u8 *, u8, s32, u32, u32, u32);
+extern void sub_80694C0(struct Entity *, s32, s32, u32);
+extern void sub_804535C(struct Entity *, u32);
+extern void sub_804AC20(struct Position *);
+extern void sub_807EC28(u32);
+extern void sub_806A5B8(struct Entity *);
+
+extern u32 gUnknown_80F5FAC[];
+extern u8 *gUnknown_80F9C4C[];
+extern u8 *gUnknown_80F9C70[];
+extern u8 *gUnknown_80F9C8C[];
+extern u8 *gUnknown_80F9CBC[];
+extern u8 gAvailablePokemonNames[];
+extern u8 *gUnknown_80FCAC0[];
+
+
+void sub_807E1A0(struct Entity *pokemon,struct Entity *target,u8 param_3,s16 param_4,s32 param_5)
+{
+  struct EntityInfo *targetInfo;
+  u8 flag;
+  struct Move move;
+  s32 newHP;
+  s32 param_4_s32;
+
+  param_4_s32 = param_4;
+
+  if (EntityExists(target)) {
+    targetInfo = target->info;
+    InitPokemonMove(&move, MOVE_REGULAR_ATTACK);
+    if (!sub_80571F0(target,&move)) {
+      if (!targetInfo->isNotTeamMember) {
+        newHP = targetInfo->HP;
+        newHP /= 2;
+        if (MonsterIsType(target, TYPE_FIRE)) {
+          newHP /= 2;
+        }
+        if (newHP < 1) {
+          newHP = 1;
+        }
+      }
+      else {
+        newHP = param_5;
+        if (MonsterIsType(target, TYPE_FIRE)) {
+           newHP /= 2;
+        }
+      }
+      sub_806F370(pokemon,target,newHP,0,&flag,param_3,param_4_s32,0,0,0);
+    }
+  }
+}
+
+void sub_807E254(struct Entity *pokemon,struct Entity *target)
+{
+  struct Position pokemonPos;
+  struct Position targetPos;
+  
+
+  if (HasAbility(pokemon, ABILITY_SUCTION_CUPS)) {
+    SetMessageArgument(gAvailablePokemonNames,pokemon,0);
+    sub_80522F4(pokemon,target,*gUnknown_80FCAC0); // $m0 is anchored! It won't switch places
```

Comment:
The functions added to this file are not related to weather. Let's move them to a new file.

Reply:
Moved